test/unit/model/findandcount.test.js

```diff
@@ -0,0 +1,46 @@
+'use strict';
```

Comment:
This test is confusing due to mocks and I am not very happy with `onPossiblyUnhandledRejection`

Why dont we add an integration test instead.